starter-code/src/components/CountryDetail.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react";
+
+import { Link } from "react-router-dom";
+
+class CountryDetail extends Component {
+  state = {};
+
+  render() {
+    // (if no state is set (directly loading url) then update the state first)
+    if (!this.props.country) {
+      this.props.setNewCountry(this.props.match.params.id);
+      return null;
+    }
+
+    return (
+      <div>
+        <h1>{this.props.country.name.official}</h1>
+        <table className="table">
+          <tbody>
+            <tr>
+              <td>Capital</td>
+              <td>{this.props.country.capital}</td>
+            </tr>
+            <tr>
+              <td>Area</td>
+              <td>{this.props.country.area} km²</td>
+            </tr>
+            <tr>
+              <td>Borders</td>
+              <td>
+                {this.props.country.borders && (
+                  <ul>
+                    {this.props.country.borders.map(border => {
+                      return (
+                        <li key={border}>
+                          <Link
+                            to={`/country/${border}`}
+                            onClick={() => {
+                              this.props.setNewCountry(border);
```

Comment:
I would style this like below. To me its a lot more readable, especially when there is already a lot of lines and indentation.  In terms of style conventions its more appropriate.  Code that conceptually belongs together should visually be positioned together.  That counts for spacing and brackets as well, the `>` should not be on its own line.
```
  <li key={border}>
        <Link to={`/country/${border}`} onClick={() => { 
              this.props.setNewCountry(border) 
           }}>
           {border}
        </Link>
  </li>
```

Reply:
Y I agree. I used "prettier" for auto format, which makes it like this